src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ClientFileMetadataFields.scala

```diff
@@ -0,0 +1,40 @@
+package uk.gov.nationalarchives.tdr.api.graphql.fields
+
+import io.circe.generic.auto._
+import sangria.macros.derive._
+import sangria.marshalling.circe._
+import sangria.schema.{Argument, Field, InputObjectType, ObjectType, fields}
+import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
+import uk.gov.nationalarchives.tdr.api.graphql.fields.FieldTypes._
+
+object ClientFileMetadataFields {
+  case class ClientFileMetadata(fileId: Long,
+                                originalPath: Option[String] = None,
+                                checksum: Option[String] = None,
+                                checksumType: Option[String] = None,
+                                lastModified: Long,
+                                createdDate: Long,
+                                fileSize: Option[BigDecimal] = None,
```

Comment:
Are some of these fields optional because they're nullable in the DB? Do you think they should be made non-nullable? I can't think of a scenario where we'd have some of these fields but not all of them.

Reply:
I was just following what is in the current DB schema. Is it not a possibility that something might fail for a file, for example filesize but other data doesn't, having the fields nullable would still let data to be stored in the DB, if that is desirable

Reply:
Hmm, I suspect it's not desirable, and it's not error handling we were planning to write any time soon, but we probably have to think about the options.

I might run this one past David and Kirsten to see what data is essential. My guess would be that it's better to fail the upload than save a file without a client-side checksum, but I'll check with them and create a new card if necessary.